src/output-json-dns.c

```diff
@@ -69,6 +132,46 @@ typedef struct LogDnsLogThread_ {
     MemBuffer *buffer;
 } LogDnsLogThread;
 
+static int DNSRRTypeEnabled(uint16_t type, uint32_t flags)
```

Comment:
would it make sense to start this func with a 'if (likely(flags == ~0)) return 1;' ? The common & default case will be that the filtering is disabled.